src/AppBundle/Controller/DangerPointController.php

```diff
@@ -72,6 +72,12 @@ public function getAllAction()
     {
         if ($this->get('security.authorization_checker')->isGranted('ROLE_ADMIN')) {
             $restresult = $this->getDoctrine()->getRepository('AppBundle:DangerPoint')->findAll();
+        } elseif (
+            $this->get('security.authorization_checker')->isGranted('ROLE_SCHOOL_ADMIN') or
```

Comment:
This check and the checks below could be extracted to a private function to make the code more concise and readable. Something like: `$this->userHasOneOfRoles('ROLE_SCHOOL_ADMIN', 'ROLE_SCHOOL_REVIEW', 'ROLE_STUDENT')`